### PR TITLE
[Enhancement]: Caching resolver dependency

### PR DESCRIPTION
**Pull request by malhotra5**
This PR partially addresses issue #143 by caching the pip installation of `openhands-resolver`

# Changes
We invoke the caching workflow only when the label `fix-me` is used; we do the following
1. Check for the latest `openhands-resolver` version available
3. Create a new cache if is doesn't exist with `openhands-resolver` version used as the key, otherwise restore packages from the existing cache

This means 
1. If `fix-me-experimental` is used all caching workflows are skipped; this ensures the unreleased version is installed
2. If new version of `openhands-resolver` is released, a new cache will be created since the cache key won't exist. The older cache will [automatically be discarded by Github in 7 days if its not used.
](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/caching-dependencies-to-speed-up-workflows#usage-limits-and-eviction-policy)

# Test logs
1. [Workflow log](https://github.com/malhotra5/test-repo/actions/runs/11734548480) where cache is created for the first time. 
2. [Workflow log](https://github.com/malhotra5/test-repo/actions/runs/11734618498) where cache hit occurs and packages are restored successfully from the cache. **Saves around 100 seconds of installation time**.
3. [Workflow log](https://github.com/malhotra5/test-repo/actions/runs/11734637138) where caching is ignored when `fix-me-experimental` label is used

### CODE REVIEW

File: .github/workflows/openhands-resolver.yml
Comment (neubig) [inline .github/workflows/openhands-resolver.yml:56]: Overall this is awesome!

Quick question: do you know if this if statement evaluates to true when this workflow is invoked through the external call in `examples/github-resolver.yml`?

Comment (malhotra5) [reply .github/workflows/openhands-resolver.yml:56]: Yup! Here is an example [workflow](https://github.com/malhotra5/test-repo/actions/runs/11736068442/job/32694720230) log.

I referenced the [workflow on PR externally](https://github.com/malhotra5/test-repo/actions/runs/11736068442/workflow).

Comment (malhotra5) [reply .github/workflows/openhands-resolver.yml:56]: On second thought, it seems that this shouldn't be the case. Will double check to ensure this is expected behavior.

Comment (neubig) [reply .github/workflows/openhands-resolver.yml:56]: Oh, I definitely would like caching to be enabled when called from an external repo. I was just confirming that this was indeed what happened with the current code.

**Review by neubig — APPROVED** [review]
I approve as long as the response to the comment below is indeed affirmative. Also @malhotra5, thanks for all your efforts on this repo! I invited you as a maintainer, so you should be able to merge this yourself now that I've approved.

File: .github/workflows/openhands-resolver.yml
Comment (malhotra5) [reply .github/workflows/openhands-resolver.yml:56]: Ah I thought maybe reusable workflows don't share `github.event` between the caller and callee unless they're called in the same repo. The test I provided above had the caller and callee in the same repo so wanted to double check this. Just confirmed that `github.event` is shared between separate repos as well, so we should be good!

**Pull request merged**
